src/main/java/teammates/storage/api/StudentsDb.java

```diff
@@ -409,44 +408,59 @@ public void updateStudent(String courseId, String email, String newName,
             if (isEmailChanged) {
                 CourseStudent newCourseStudent = new CourseStudent(newEmail, newName, newGoogleId, newComments,
                                                                    courseId, newTeamName, newSectionName);
-                newCourseStudent.setLastName(lastName);
-                newCourseStudent.setCreatedAt(courseStudent.getCreatedAt());
-                if (keepUpdateTimestamp) {
-                    newCourseStudent.setLastUpdate(courseStudent.getUpdatedAt());
-                }
-
-                try {
-                    createStudent(new StudentAttributes(newCourseStudent), hasDocument);
-                } catch (EntityAlreadyExistsException e) {
-                    StudentAttributes existingStudent = (StudentAttributes) e.existingEntity;
-                    String error = ERROR_UPDATE_EMAIL_ALREADY_USED
-                            + existingStudent.getName() + "/" + existingStudent.getEmail();
-                    throw new InvalidParametersException(error);
-                }
-
-                deleteStudent(courseId, email);
-
+                updateStudentChangeEmail(newCourseStudent, lastName, courseStudent, hasDocument,
+                                         keepUpdateTimestamp, courseId, email);
             } else {
-                courseStudent.setName(newName);
-                courseStudent.setLastName(lastName);
-                courseStudent.setComments(newComments);
-                courseStudent.setGoogleId(newGoogleId);
-                courseStudent.setTeamName(newTeamName);
-                courseStudent.setSectionName(newSectionName);
-
-                if (hasDocument) {
-                    putDocument(new StudentAttributes(courseStudent));
-                }
-
-                // Set true to prevent changes to last update timestamp
-                courseStudent.keepUpdateTimestamp = keepUpdateTimestamp;
+                updateStudentDetails(newName, newTeamName, newSectionName, newGoogleId,
+                                     newComments, hasDocument, keepUpdateTimestamp, courseStudent, lastName);
             }
         }
 
         log.info(Const.SystemParams.COURSE_BACKUP_LOG_MSG + courseId);
         getPm().close();
     }
 
+    @SuppressWarnings("PMD.PreserveStackTrace")
+    public void updateStudentChangeEmail(
```

Comment:
Hmm yup, looking at it again it does indeed reflect the behaviour more accurately